CI run against latest deps is failing

See https://github.com/matrix-org/synapse/actions/runs/2991362357

```
Run poetry run mypy
synapse/api/filtering.py:142: error: Value of type variable "_F" of function cannot be "Callable[[str], RoomID]"  [type-var]
synapse/api/filtering.py:147: error: Value of type variable "_F" of function cannot be "Callable[[str], UserID]"  [type-var]
```

--- a/changelog.d/13712.misc
+++ b/changelog.d/13712.misc
@@ -1,0 +1,1 @@
+Fix typechecking with latest types-jsonschema.

--- a/synapse/api/filtering.py
+++ b/synapse/api/filtering.py
@@ -140,13 +140,13 @@ USER_FILTER_SCHEMA = {
 
 
 @FormatChecker.cls_checks("matrix_room_id")
-def matrix_room_id_validator(room_id_str: str) -> RoomID:
-    return RoomID.from_string(room_id_str)
+def matrix_room_id_validator(room_id_str: str) -> bool:
+    return RoomID.is_valid(room_id_str)
 
 
 @FormatChecker.cls_checks("matrix_user_id")
-def matrix_user_id_validator(user_id_str: str) -> UserID:
-    return UserID.from_string(user_id_str)
+def matrix_user_id_validator(user_id_str: str) -> bool:
+    return UserID.is_valid(user_id_str)
 
 
 class Filtering:

